File: frontend/lib/features/auth/models/register_patient_request.dart
class RegisterPatientRequest {
  final String email;
  final String name;
  final String surname;
  final int dni;
  final String dateOfBirth; // YYYY-MM-DD
  final String address;
  final int number;

  RegisterPatientRequest({
    required this.email,
    required this.name,
    required this.surname,
    required this.dni,
    required this.dateOfBirth,
    required this.address,
    required this.number,
  });

  Map<String, dynamic> toJson() => {
    "email": email,
    "name": name,
    "surname": surname,
    "dni": dni,
    "dateOfBirth": dateOfBirth,
    "address": address,
    "number": number,
  };
}

File: backend/src/main/java/com/agustin/backend_dialysis_record/dto/auth/RegisterPatientRequest.java
package com.agustin.backend_dialysis_record.dto.auth;

import java.time.LocalDate;
import jakarta.validation.constraints.*;

public record RegisterPatientRequest(
        @NotBlank
        @Email
        @Size(max = 254)
        String email,

        // password eliminado: la identidad es gestionada por Supabase Auth

        @NotBlank
        @Size(max = 60)
        String name,

        @NotBlank
        @Size(max = 60)
        String surname,

        @NotNull
        @Min(1_000_000)
        @Max(99_999_999)
        Integer dni,

        @NotNull
        @Past
        LocalDate dateOfBirth,

        @NotBlank
        @Size(max = 120)
        String address,

        @Min(1)
        Long number
) {}
